app/services/ui_notifications/url_resolver.rb

```diff
@@ -0,0 +1,52 @@
+module UINotifications
+  # interpolates blueprint strings and links for notifications
+  class URLResolver
+    # needed in order to resolve rails urls for notifications
+    include Rails.application.routes.url_helpers
+    def initialize subject, message, actions
+      @subject = subject
+      @raw_message = message
+      @raw_actions = actions
+    end
+
+    def message
+      parse_string(raw_message)
+    end
+
+    def actions
+      return if raw_actions.try(:links).nil?
+      raw_actions[:links].map do |link|
+        if link.has_key? :href
+          link
+        elsif link.has_key? :path_method
+          validate_link(link)
+          parse_link(link[:path_method], link[:title])
+        end
+      end
+    end
+
+    private
+
+    attr_reader :subject, :raw_message, :raw_actions
+    def parse_string string
+      string % { :subject => subject }
+    end
+
+    def parse_link path_method, title
+      {
+        href: public_send(path_method, subject),
```

Comment:
It will fail for index paths like `hosts_path`, we can't pass the subject to such paths.

``` ruby
hosts_path(Host.first)
#=> "/hosts.tommy-havier.mydomain.com"
```